src/FileGenerator/InvoicePdfFileGenerator.php

```diff
@@ -68,10 +68,34 @@ public function generateFile(InvoiceInterface $invoice): string
                 'companyData' => $this->companyDataResolver->resolveCompanyData(),
             ]
         );
-        $path = $this->filesPath . '/' . (string) $invoice->getId() . bin2hex(random_bytes(6)) . '.pdf';
-
+        $filename = $this->getInvoiceFilename($invoice);
+        $path = $this->filesPath . DIRECTORY_SEPARATOR . $filename;
+        
         $this->pdfFileGenerator->generateFromHtml($html, $path);
 
-        return $path;
+        return $filename;
+    }
+
+    /**
+     * @param InvoiceInterface $invoice
+     * @return string Returns an explicit invoice file name
+     */
+    protected function getInvoiceFilename(InvoiceInterface $invoice): string
```

Comment:
Can we move it to another service? As this is a vendor plugin, I think it's worth to allow decorating/overriding this method. If this is too much for you, please change it's visibility to private, as the service is a final class and extending services in the DI app is not the best practice. Please, move it also to the bottom of the class. We're ordering methods by their visibility, so public comes above protected and private are the last ones 🙂